stages/org.osbuild.squashfs

```diff
@@ -76,8 +76,9 @@ def main(inputs, output_dir, options):
         method = compression["method"]
         opts = compression.get("options", {})
         cmd += ["-comp", method]
-        for opt, val in opts.items():
-            cmd += [f"-X{opt}", val]
+        if method == "xz":
```

Comment:
The compression level should also be available for `gzip`

Reply:
I agree.
Still I am confused because, in the man pages of `mksquashfs` the options `-Xbcj` is not available for ègzip`, but it is the only one listed in the schema at the top of the file.

```
   Compressors available and compressor specific options
       gzip (default)

       -Xcompression-level compression-level
           compression-level should be 1 .. 9 (default 9)

       -Xwindow-size window-size
           window-size should be 8 .. 15 (default 15)

       -Xstrategy strategy1,strategy2,...,strategyN
           Compress using strategy1,strategy2,...,strategyN in turn and choose the best compression. Available strategies: default, filtered, huffman_only, run_length_encoded and fixed

       lzmz (no options) (no kernel support)

       lzo

       -Xalgorithm algorithm
           Where algorithm is one of: lzo1x_1, lzo1x_1_11, lzo1x_1_12, lzo1x_1_15 or lzo1x_999. (default lzo1x_999)

       -Xcompression-level compression-level
           compression-level should be 1 .. 9 (default 8)

       lz4

       -Xhc
           Compress using LZ4 High Compression

       xz

       -Xbcj filter1,filter2,...,filterN
           Compress using filter1,filter2,...,filterN in turn (in addition to no filter), and choose the best compression. Available filters: x86, arm, armthumb, powerpc, sparc, ia64.

       -Xdict-size DICT_SIZE
           Use  DICT_SIZE  as the XZ dictionary size. The dictionary size can be specified as a percentage of the block size, or as an absolute value. The dictionary size must be less than or equal to the
           block size and 8192 bytes or larger. It must also be storable in the xz header as either 2^n or as 2^n+2^(n+1). Example dict-sizes are 75%, 50%, 37.5%, 25%, or 32K, 16K, 8K etc.

       zstd

       -Xcompression-level <compression-level>
           <compression-level> should be 1 .. 22 (default 15)
```

Reply:
You're right.  It seems the stage supports the options it needs for all compression algorithms but doesn't validate whether the options are correct for the algo.  I suppose it's up to the caller to set them properly and the stage will just pass them through to the command and fail if they're invalid (or does it ignore them?).

Reply:
I don't know if we should implement some kind of verification or not. I we do, it might be cumbersome do validate all the corner cases. For instance, some of the `-X `options are taking a parameter behind them, and others are just a flag. Which makes our for loop not working in the case of `lz4` for instance. Also it seems redundant and easily breakable if the underlying program is updating its api.

Maybe the best thing to do would be to drop the logic associated with the compression options and just let the user enter whatever he wants in a string. And I don't know if that's ideal, because it looks like a security mistake to let a user execute whatever string on the system.

Reply:
I think the current state is fine.  We support a limited set of options (like we do with almost every stage) and bad combinations of options are only validated if it's straightforward.  Like you said, it's hard to cover all cases and we might be locking out legitimate cases if we try to verify.  Arbitrary options aren't necessary.  Keep it as it is now :)